framework/src/utils/MatrixTools.C

```diff
@@ -43,7 +45,14 @@ void
 inverse(std::vector<PetscScalar> & A, unsigned int n)
 {
   mooseAssert(n >= 1, "MatrixTools::inverse - n (leading dimension) needs to be positive");
-  mooseAssert(n == static_cast<int>(n), "MatrixTools::inverse - n (leading dimension) too large");
+
+#ifndef NDEBG
+  constexpr auto N = CHAR_BIT * sizeof(unsigned int);
+
+  std::bitset<N> most_significant_bit_test(n);
```

Comment:
Agreed - my brain immediately went to "bits". Didn't really think about looking or trying for another way. It's moot now. I already fixed it.